compiler/rustc_const_eval/src/interpret/intrinsics.rs

```diff
@@ -258,6 +258,7 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             sym::copy => {
                 self.copy_intrinsic(&args[0], &args[1], &args[2], /*nonoverlapping*/ false)?;
             }
+            sym::is_constant => self.write_scalar(Scalar::from_bool(true), dest)?,
```

Comment:
This implementation says that if I do
```rust
fn main() {
  let x: i32 = read_from_std();
  assert!(is_constant(x));
}
```
then this must always return true. That cannot be right.

The code is isn't just used for CTFE, it is also used for Miri, and must be an accurate reflection of the *run-time semantics* of Rust.

Reply:
The semantics are less "is_const" and more "should_take_code_path_that_is_more_efficient_if_arg_is_const"